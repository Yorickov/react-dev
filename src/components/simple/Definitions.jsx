import React from 'react';
import { uniqueId } from 'lodash';

const Definitions = ({ data }) => {
  const renderDef = (array) => (
    <dl>
      {array.map(({ dt, dd }) => (
        <React.Fragment key={uniqueId()}>
          <dt>{dt}</dt>
          <dd>{dd}</dd>
        </React.Fragment>
      ))}
    </dl>
  );

  return data.length > 0 && renderDef(data);
};

export default Definitions;
