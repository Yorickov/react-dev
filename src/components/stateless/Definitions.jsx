import React from 'react';
import { uniqueId } from 'lodash';
import PropTypes from 'prop-types';

const Definitions = ({ data }) => {
  const renderDef = (array) => (
    <dl>
      {array.map(({ dt, dd }) => (
        <React.Fragment key={uniqueId()}>
          <dt>{dt}</dt>
          <dd>{dd}</dd>
        </React.Fragment>
      ))}
    </dl>
  );

  return data.length > 0 && renderDef(data);
};

Definitions.propTypes = {
  data: PropTypes.array,
};

export default Definitions;
