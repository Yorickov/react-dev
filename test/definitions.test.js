import React from 'react';
import Definitions from '../src/components/simple/Definitions.jsx';

it('Definitions 1', () => {
  const definitions = [
    { dd: 'term one', dt: 'description one' },
    { dd: 'term two', dt: 'description two' },
  ];
  const wrapper = shallow(<Definitions data={definitions} />);

  expect(wrapper.find('dd')).toHaveLength(2);
  expect(wrapper.render()).toMatchSnapshot();
});

it('Definitions 2', () => {
  const definitions = [];
  const wrapper = shallow(<Definitions data={definitions} />);

  expect(wrapper.find('dd')).toHaveLength(0);
  expect(wrapper.render()).toMatchSnapshot();
});
