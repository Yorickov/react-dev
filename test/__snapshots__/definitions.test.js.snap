// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definitions 1 1`] = `
<dl>
  <dt>
    description one
  </dt>
  <dd>
    term one
  </dd>
  <dt>
    description two
  </dt>
  <dd>
    term two
  </dd>
</dl>
`;

exports[`Definitions 2 1`] = `null`;
